dashboard/src/modules/components/OverviewComponent/ExpiringSoonComponent.jsx

```diff
@@ -1,19 +1,65 @@
 import "./index.less";
 
+import {
+  DEFAULT_PER_PAGE_EXPIRING,
+  START_PAGE_NUMBER,
+} from "assets/constants/overviewConstants";
 import { List, ListItem } from "@patternfly/react-core";
+import React, { useCallback, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
 
-import React from "react";
-import { useSelector } from "react-redux";
+import { RenderPagination } from "./common-component";
+import { setExpiringRows } from "actions/overviewActions";
 
 const ExpiringSoonComponent = () => {
-  const { expiringRuns } = useSelector((state) => state.overview);
+  const dispatch = useDispatch();
+  const { expiringRuns, initExpiringRuns } = useSelector(
+    (state) => state.overview
+  );
+
+  /* Pagination */
+  const [perPage, setPerPage] = useState(DEFAULT_PER_PAGE_EXPIRING);
+  const [page, setPage] = useState(START_PAGE_NUMBER);
 
+  const onSetPage = useCallback(
+    (_evt, newPage, _perPage, startIdx, endIdx) => {
+      setPage(newPage);
+      dispatch(setExpiringRows(expiringRuns.slice(startIdx, endIdx)));
+    },
+    [dispatch, expiringRuns]
+  );
+  const perPageOptions = [
+    { title: "10", value: 10 },
+    { title: "15", value: 15 },
+    { title: "20", value: 20 },
+  ];
+  const onPerPageSelect = useCallback(
+    (_evt, newPerPage, newPage, startIdx, endIdx) => {
+      setPerPage(newPerPage);
+      setPage(newPage);
+      dispatch(setExpiringRows(expiringRuns.slice(startIdx, endIdx)));
+    },
+    [dispatch, expiringRuns]
+  );
+  /* Pagination */
```

Comment:
I'm not sure what this code comment is supposed to be telling me...can we either expand or remove it?  (Its twin at line 20 seems to be introducing the definitions there, but, given that they all have "page" in their names, I'm not sure that the comment adds much value.  (Also, didn't we settle on using `//`-style line comments instead of `/* ... */` ones?)

Reply:
Modified the  comment style